core/src/main/java/hudson/cli/ClientAuthenticationCache.java

```diff
@@ -110,9 +110,12 @@ public Authentication get() {
      */
     @VisibleForTesting
     String getPropertyKey() {
-        String url = Jenkins.getActiveInstance().getRootUrl();
+        Jenkins j = Jenkins.getActiveInstance();
+        String url = j.getRootUrl();
         if (url!=null)  return url;
-        return Secret.fromString("key").getEncryptedValue();
+        
+        LOGGER.log(Level.WARNING, "The instance is not configured using a rootUrl, the key that represents your instance will not be stable");
```

Comment:
Well, actually, it will be _more_ stable, since `legacyInstanceId` is guaranteed to be consistent for the lifetime of this Jenkins service, whereas the root URL is subject to reconfiguration. So this warning is misleading and should probably just be dropped.

Reply:
Correction proposed in [#3117](https://github.com/jenkinsci/jenkins/pull/3117)